extensions/wrapper/clients/typescript-client/src/EdcClient.ts

```diff
@@ -20,12 +22,30 @@ export interface EdcClient {
  * @param opts opts
  */
 export function buildEdcClient(opts: EdcClientOptions): EdcClient {
+    let fetchWrapper: OAuth2Fetch | null = null;
+    if (opts.oAuth2ClientCredentials) {
+        const client = new OAuth2Client({
+            server: opts.oAuth2ClientCredentials.serverUrl,
+            tokenEndpoint: opts.oAuth2ClientCredentials.tokenEndpoint,
+            clientId: opts.oAuth2ClientCredentials.clientId,
+            clientSecret: opts.oAuth2ClientCredentials.clientSecret,
+        });
+        // Wrapper for fetch that automatically sets Authorization: Bearer headers and refreshes the token when needed
+        fetchWrapper = new OAuth2Fetch({
+            client: client,
+            getNewToken: async () => {
+                return client.clientCredentials();
+            },
+        });
+    }
```

Comment:
let's maybe move this all to a method to keep the entrypoint "buildEdcClient" clean. seeing that you added a directory "oauth2" you could build it there in an abstracted way `export function fetchApiWithClientCredentials(config: OAuth2ClientCredentials): FetchAPI`